edx_django_utils/monitoring/internal/middleware.py

```diff
@@ -285,6 +286,19 @@ def __call__(self, request):
 
         response = self.get_response(request)
 
+        # .. setting_name: COOKIE_PREFIXES_TO_REMOVE
+        # .. setting_default: []
+        # .. setting_description: A list of cookie prefixes. Any cookie starting with a prefix in this list
+        # will be manually removed from responses (i.e. set to expire in the past)
+        deprecated_cookie_prefixes = getattr(settings, "COOKIE_PREFIXES_TO_REMOVE", [])
+        yesterday = datetime.utcnow() - timedelta(days=1)
+        for cookie_name in request.COOKIES.keys():
+            for deprecated_cookie_prefix in deprecated_cookie_prefixes:
```

Comment:
I'm a little concerned about what might happen if `deprecated_cookie_prefixes` is accidentally set to a string, e.g. an attempted single-tuple like `("testing",)`. This would delete every cookie starting with the letter t, e, s, i, n, or g.

Maybe the setting can be read in the `__init__` method of the middleware, and checked to make sure it's a list of strings, and then assigned to an instance variable? And if not, emit a warning and skip this whole block.

Reply:
(edited example)

Reply:
Making it a list of (name, domain) tuples mitigates this issue since it will error trying to unpack two values. I guess we can still check if it's an array to avoid some other strange error

Reply:
[fixed]